corehq/motech/repeaters/expression/repeaters.py

```diff
@@ -63,6 +84,35 @@ def get_url(self, repeat_record):
             )
         return base_url
 
+    def handle_response(self, response, repeat_record):
+        super().handle_response(response, repeat_record)
+        if self.update_case_filter_expression and is_response(response):
```

Comment:
nit: just checking about the function is_response here.
I understand status_code but should a response from an external system have a "reason" attribute?

Reply:
> nit: just checking about the function is_response here. I understand status_code but should a response from an external system have a "reason" attribute?

I'm reusing an existing utility. I can't really speak to why those two fields were selected as indicators of a `Response` class. Having said that, I would certainly expect that if it is a `requests.Response` class then it will certainly have those two attributes and the combination of the two is likely rare so it seems a reasonable choice.

Reply:
Very late to this party, sorry, but figured I'd close this loop. `is_response()` ensures that we have `status_code` and `reason` because we need them [when we create a `RepeatRecordAttempt` here](https://github.com/dimagi/commcare-hq/blob/526a5572affa63b443a0f19646e063fbff1e010f/corehq/motech/repeaters/models.py#L1038). We use them to populate `RepeatRecordAttempt.message`, which is available in the Repeat Records Report under the "Responses" button.

Reply:
thanks @kaapstorm